sdks/python/apache_beam/runners/sdf_utils.py

```diff
@@ -244,3 +247,47 @@ def get_estimator_state(self):
         return None
 
     return _NoOpWatermarkEstimator()
+
+
+class DeduplicationByUniqueId(ptransform.PTransform):
+  def __init__(self, windowfn=GlobalWindows(), trigger=None):
```

Comment:
I think if the user wants to specify things in terms of windows and triggers, it would be more natural to manually do windowing before this operation. Instead, perhaps higher-level semantic information could be provided (namely, over what interval should the deduplication occur), and windowing/triggering should be used to accomplish this.